From: Bhavana Nagendra <bnagendr@redhat.com>
Date: Thu, 13 Mar 2008 12:57:34 -0400
Subject: [sound] HDMI device IDs for AMD ATI chipsets
Message-id: 20080313155735.10231.51727.sendpatchset@localhost.localdomain
O-Subject: [RHEL5.2 PATCH]: HDMI device IDs for AMD ATI chipsets
Bugzilla: 435658

Resolves BZ 435658

Add HDMI support for AMD ATI chipsets to allow HDMI to work on all known
chipset shipping configurations. This patch adds the necessary device IDs.
This patch is based on kernel-2.6.18-84.el5. The similar patch has been
accepted by ALSA in its hg tree (AMD chipset team).

Upstream link:
http://git.kernel.org/?p=linux/kernel/git/perex/alsa.git;a=commitdiff;h=f98d4a4a2ae926238a6dc908382dc6a963c44fe5;hp=864cae534636695febd30659dde9f4f223f3e244

Brew build:
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=1203579

Testing:
This patch has been tested by AMD chipset team on Shiner SB700 A12 board,
along with the BZ 428963 support (that is already in QA).  The "aplay" &
"speaker-test" work well.  RHEL5.2 is missing the modprobe change (BZ 436257)
and they had to make the changes to modprobe.conf to succesfully test patch.

Please ACK.

Acked-by: Brian Maly <bmaly@redhat.com>
Acked-by: Pete Zaitcev <zaitcev@redhat.com>

diff --git a/sound/pci/hda/hda_intel.c b/sound/pci/hda/hda_intel.c
index 9084274..392548e 100644
--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -1777,6 +1777,9 @@ static struct pci_device_id azx_ids[] = {
 	{ 0x1002, 0xaa20, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI RV635 HDMI */
 	{ 0x1002, 0xaa28, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI RV620 HDMI */
 	{ 0x1002, 0xaa30, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI RV770 HDMI */
+	{ 0x1002, 0xaa38, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI HDMI */
+	{ 0x1002, 0xaa40, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI HDMI */
+	{ 0x1002, 0xaa48, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ATIHDMI }, /* ATI HDMI */
 	{ 0x1106, 0x3288, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_VIA }, /* VIA VT8251/VT8237A */
 	{ 0x1039, 0x7502, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_SIS }, /* SIS966 */
 	{ 0x10b9, 0x5461, PCI_ANY_ID, PCI_ANY_ID, 0, 0, AZX_DRIVER_ULI }, /* ULI M5461 */
